test/javabind.cpp

```diff
@@ -493,6 +493,14 @@ std::ostream& operator<<(std::ostream& os, FooBar value)
     }
 }
 
+struct StringHash
+{
+    std::size_t operator()(const std::string& str) const
+    {
+        return str.empty() ? 0 : str.at(0);
```

Comment:
Even though this is only a test, we might want to use a more realistic hash function. I would simply call `std::hash<std::string>` in the body, and e.g. flip all bits of the result.

Reply:
Changed to return std::hash with the bits flipped